flow/Makefile

```diff
@@ -651,10 +651,41 @@ $(eval $(call do-copy,3_place,3_5_place_dp.odb,))
 
 $(eval $(call do-copy,3_place,2_floorplan.sdc,,.sdc))
 
+# This target does not produce a non-zero exit code if detailed or global
+# placement fails. In this case $(RESULTS_DIR)/3_place.odb and $(RESULTS_DIR)/3_place.sdc
+# are produced, such that an artifact can be created for inspection purposes.
+#
+# The check-place target subsequently will fail if
+# placement did not complete successfully, because $(RESULTS_DIR)/place.ok
+# is set to 0.
+#
+# This means that build systems, such as Bazel, that require a non-zero
+# exit code and the same artifacts to be produced every time, or it
+# will not publish the artifacts, can use this target.
 .PHONY: do-place
 do-place:
 	mkdir -p $(LOG_DIR) $(REPORTS_DIR)
-	$(UNSET_AND_MAKE) do-3_1_place_gp_skip_io do-3_2_place_iop do-3_3_place_gp do-3_4_place_resized do-3_5_place_dp do-3_place do-3_place.sdc
+	echo >$(RESULTS_DIR)/place.ok 0
+	$(UNSET_AND_MAKE) do-3_1_place_gp_skip_io do-3_2_place_iop
+	$(UNSET_AND_MAKE) WRITE_ON_FAIL=1 do-3_3_place_gp ; \
+	if [ $$? -ne 0 ]; then \
+		cp $(RESULTS_DIR)/3_3_place_gp.odb $(RESULTS_DIR)/3_place.odb ; \
+		$(UNSET_AND_MAKE) do-3_place.sdc ; \
+		exit 1 ; \
+	fi
+	$(UNSET_AND_MAKE) do-3_4_place_resized
+	$(UNSET_AND_MAKE) WRITE_ON_FAIL=1 do-3_5_place_dp
+	if [ $$? -ne 0 ]; then \
+		cp $(RESULTS_DIR)/3_5_place_dp.odb $(RESULTS_DIR)/3_place.odb ; \
+		$(UNSET_AND_MAKE) do-3_place.sdc ; \
+		exit 1 ; \
+	fi
+	$(UNSET_AND_MAKE) do-3_place do-3_place.sdc
+	echo >$(RESULTS_DIR)/place.ok 1
```

Comment:
@maliberty This file contains success/failure

Reply:
Make will not read the contents of the file on the next invocation and will think this step succeeded.

Reply:
Indeed:

1. do-xxxx is not compatible w normal make target checking, you have to pick one or the other. This is fine.
2. I need to add check-place to do-cts and check-route to do-finish